components/micronova.rst

```diff
@@ -0,0 +1,201 @@
+MicroNova based pellet stove
+============================
+
+.. seo::
+    :description: Instructions for setting up a MicroNova board based pellet stove in ESPHome.
+
+The MicroNova component allows you to integrate a pellet stove with a MicroNova board in ESPHome.
+It uses :ref:`UART <uart>` for communication.
+
+The :ref:`UART <uart>` must be configured with a baud rate 1200, 8 data bits, 2 stop bits, no parity, no flow control. 
+
+.. warning::
+
+    The MicroNova bords come in various flavours. This code is only tested on an ExtraFlame Ketty Evo 2.0 stove. The protocol is not
+    documented but has been reverse engineerd by others. See the links below for all the info that helped me.
+    The different sensors, buttons and stove switch may require specific **memory_location** and **memory_address** parameters that 
+    match your MicroNova specific board.
+
+    Also, switching your stove on or off can behave different on the various MicroNova flavours.
+
+    Use this integration at your own risk.
+
+Connecting your stove
+---------------------
+
+Most MicroNova based pellet stoves have a serial output. In most cases this output has 4 pins: GND, 5v, 20V and DATA.
+
+.. figure:: images/micronova_serial.png
+    :align: center
+    :width: 50.0%
+
+.. figure:: images/micronova_serial_layout.png
+    :align: center
+    :width: 50.0%
+
+You will have to build a simple circuit to interface with your stove. It is based on optocouplers for galvanic separation and logic 
+level shifting between 5v and 3V3.
+
+.. figure:: images/micronova_optocouplers.png
+    :align: center
+    :width: 100.0%
+
+    Optocoupler interface circuit (credit: philibertc)
+
+
+
+See the references below for all the details about te circuit.
+
+You can use the 5V output from the stove to power the ESP module, but you will have to put a voltage regulator in between to
+get 3v3.
+
+MicroNova configuration
+-----------------------
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      tx_pin: 5
+      rx_pin: 4
+      baud_rate: 1200
+      stop_bits: 2
+
+    micronova:
+      enable_rx_pin: 4
+      update_interval: 20s
```

Comment:
We should not suggest values outside the defaults here and also the PIN should be set to the special value of `REPLACEME` so that when people do copy/paste eexamples, it forces them to change it to match their set up.
```suggestion
    micronova:
      enable_rx_pin: REPLACEME
```

Reply:
Done